### DISCUSSION TITLE
Getting element DOM property (like innerHTML)

### DISCUSSION THREAD

**Discussion by ghost** [php-webdriver/php-webdriver#928] [Q&A] [answered]
Is there any way to get the innerHTML the native way?

This works, but I'm looking for something like `->getHTML()`:
`$driver->executeScript('return document.querySelector(".selector").innerHTML')`

I've tried this with no success:
`$driver->findElement(
     WebDriverBy::cssSelector('div.slider-wrap:nth-child(3)')
)->getAttribute('innerHTML');`

I'm using this docker image: `selenium/standalone-firefox:4.0.0-rc-1-prerelease-20210804`

`$driver = RemoteWebDriver::create($fullUrl, DesiredCapabilities::firefox());`

**Comment by OndraM** [comment, marked as answer]
Hi, this is indeed a missing feature in out W3C webdriver protocol implementation. I created issue #929 with further information. Thanks for reporting this!
